.github/workflows/osdk_test.yml

```diff
@@ -31,3 +31,19 @@ jobs:
       - name: Unit test
         id: unit_test
         run: cd osdk && cargo build && RUSTUP_HOME=/root/.rustup cargo test
+  
+  # Test OSDK in the same environment as described in the OSDK User Guide in the Asterinas Book.
+  osdk_doc_env_test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/osdk:0.4.0
+    steps:
+      - run: echo "Running in asterinas/osdk:0.4.0"
+
+      - uses: actions/checkout@v4
+
+      - name: Unit test
+        id: unit_test
+        run: cd osdk && cargo build && RUSTUP_HOME=/root/.rustup cargo test
+    
+    # TODO: test the demos in Asterinas Book
```

Comment:
One way to do this is write the demos in the Asterinas Book as doctests, unit tests, or examples in the OSDK crate (or even a separate crate for a sample OSDK project). Run these tests with `cargo test` in CI. And then, use [MdBook's include](https://rust-lang.github.io/mdBook/format/mdbook.html#including-files) feature to include the corresponding source code. This way, the Book's source code and testable code are always kept in sync.

Reply:
The demos are added to OSDK integration test.